rotkehlchen/api/rest.py

```diff
@@ -5009,3 +5010,60 @@ def get_event_details(self, identifier: int) -> Response:
             return api_response(wrap_in_fail_result('No details found'), status_code=HTTPStatus.NOT_FOUND)  # noqa: E501
 
         return api_response(_wrap_in_ok_result(details), status_code=HTTPStatus.OK)
+
+    def add_evm_transaction_by_hash(
+            self,
+            async_query: bool,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> Response:
+        """Adds an evm transaction to the DB and associate it with the address provided.
+        If successful, the transaction is then decoded.
+        """
+        if async_query is True:
+            return self._query_async(
+                command=self._add_evm_transaction_by_hash,
+                evm_chain=evm_chain,
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+
+        response = self._add_evm_transaction_by_hash(
+            evm_chain=evm_chain,
+            tx_hash=tx_hash,
+            associated_address=associated_address,
+        )
+        status_code = _get_status_code_from_async_response(response)
+        result_dict = {'result': response['result'], 'message': response['message']}
+        return api_response(process_result(result_dict), status_code=status_code)
+
+    def _add_evm_transaction_by_hash(
+            self,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> dict[str, Any]:
+        evm_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain)
+        try:
+            transaction, tx_receipt = evm_manager.transactions.add_transaction_by_hash(
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+        except (KeyError, DeserializationError, RemoteError, sqlcipher.IntegrityError, InputError) as e:  # pylint: disable=no-member  # noqa: E501
+            if isinstance(e, InputError):
+                status_code = HTTPStatus.NOT_FOUND
+            elif isinstance(e, sqlcipher.IntegrityError):  # pylint: disable=no-member
+                status_code = HTTPStatus.CONFLICT
+            else:
+                status_code = HTTPStatus.BAD_GATEWAY
+
+            return wrap_in_fail_result(
+                message=(
+                    f'Unable to add transaction with hash {tx_hash.hex()} for chain {evm_chain} and '  # noqa: E501
+                    f'associated address {associated_address} due to {str(e)}'
```

Comment:
Don't add noqa if you can split it into 2 lines equally
```suggestion
                    f'Unable to add transaction with hash {tx_hash.hex()} for chain  '
                    f'{evm_chain} and associated address {associated_address} due to {str(e)}'
```